File: gitdot-web/packages/api/src/endpoint/organization/create-organization.ts
import { z } from "zod";
import { OrganizationResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const CreateOrganizationRequest = z.object({
  readme: z.string().optional(),
});
export type CreateOrganizationRequest = z.infer<
  typeof CreateOrganizationRequest
>;

export const CreateOrganizationResponse = OrganizationResource;
export type CreateOrganizationResponse = z.infer<
  typeof CreateOrganizationResponse
>;

export const CreateOrganization = {
  path: "/organization/{org_name}",
  method: "POST",
  request: CreateOrganizationRequest,
  response: CreateOrganizationResponse,
} as const satisfies Endpoint;
export type CreateOrganization = typeof CreateOrganization;

File: gitdot-api/src/endpoint/organization/create_organization.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::OrganizationResource};

pub struct CreateOrganization;

impl Endpoint for CreateOrganization {
    const PATH: &'static str = "/organization/{org_name}";
    const METHOD: http::Method = http::Method::POST;

    type Request = CreateOrganizationRequest;
    type Response = CreateOrganizationResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct CreateOrganizationRequest {
    pub readme: Option<String>,
}

pub type CreateOrganizationResponse = OrganizationResource;
